src/Nox.Types/Types/Year/Year.cs

```diff
@@ -1,9 +1,38 @@
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="Year"/> type and value object.
+/// </summary>
+public sealed class Year : ValueObject<ushort, Year>
+{
     /// <summary>
-    /// Represents a Nox <see cref="Year"/> type and value object.
+    /// The minimum valid year value.
     /// </summary>
-    /// <remarks>Placeholder, needs to be implemented</remarks>
-    public sealed class Year : ValueObject<ushort, Year>
+    private const ushort MinYearValue = 1;
```

Comment:
We need to figure out the proper Min and Max  year, because 1 and 9999 do not really constrain valid values

We need to validate this with @andresharpe  or @jan-schutte

Reply:
The min and max values are C# DateTime min and max values and make sense as default values, but we might expose them through YearTypeOptions so that you can set them for your attributes still in this range of 1-9999.
![image](https://github.com/NoxOrg/Nox.Types/assets/131167535/7a1eafcb-fb52-4373-bc3f-b68cf4ef228e)

CC: @rochar @andresharpe @jan-schutte

Reply:
@denisvieira-dev according to mysql and mssql, lets constrain year to start with 1753 and max 2155

Reply:
So lets keep the default between 1 and 9999

And create a YearTypeOptions that the user can constrain according to its own domain